ee-9/source-transform/microprofile/opentracing-extension/src/main/java/org/wildfly/extension/microprofile/opentracing/TracingDeploymentProcessor.java

```diff
@@ -0,0 +1,293 @@
+import static org.wildfly.extension.microprofile.opentracing.SubsystemDefinition.TRACER_CAPABILITY;
+import static org.wildfly.extension.microprofile.opentracing.TracingExtensionLogger.ROOT_LOGGER;
+import static org.wildfly.microprofile.opentracing.smallrye.TracerConfigurationConstants.SMALLRYE_OPENTRACING_SERVICE_NAME;
+import static org.wildfly.microprofile.opentracing.smallrye.TracerConfigurationConstants.SMALLRYE_OPENTRACING_TRACER;
+import static org.wildfly.microprofile.opentracing.smallrye.TracerConfigurationConstants.SMALLRYE_OPENTRACING_TRACER_CONFIGURATION;
+import static org.wildfly.microprofile.opentracing.smallrye.TracerConfigurationConstants.SMALLRYE_OPENTRACING_TRACER_MANAGED;
+import static org.wildfly.microprofile.opentracing.smallrye.TracerConfigurationConstants.TRACER_CONFIGURATION;
+import static org.wildfly.microprofile.opentracing.smallrye.TracerConfigurationConstants.TRACER_CONFIGURATION_NAME;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import io.opentracing.Tracer;
+import io.opentracing.noop.NoopTracerFactory;
+import org.jboss.as.controller.capability.CapabilityServiceSupport;
+import org.jboss.as.ee.structure.DeploymentType;
+import org.jboss.as.ee.structure.DeploymentTypeMarker;
+import org.jboss.as.server.deployment.AttachmentKey;
+import org.jboss.as.server.deployment.Attachments;
+import org.jboss.as.server.deployment.DeploymentPhaseContext;
+import org.jboss.as.server.deployment.DeploymentResourceSupport;
+import org.jboss.as.server.deployment.DeploymentUnit;
+import org.jboss.as.server.deployment.DeploymentUnitProcessingException;
+import org.jboss.as.server.deployment.DeploymentUnitProcessor;
+import org.jboss.as.web.common.ServletContextAttribute;
+import org.jboss.as.web.common.WarMetaData;
+import org.jboss.as.weld.WeldCapability;
+import org.jboss.metadata.javaee.spec.ParamValueMetaData;
+import org.jboss.metadata.web.jboss.JBossWebMetaData;
+import org.jboss.metadata.web.spec.DispatcherType;
+import org.jboss.metadata.web.spec.FilterMappingMetaData;
+import org.jboss.metadata.web.spec.FilterMetaData;
+import org.jboss.metadata.web.spec.FiltersMetaData;
+import org.jboss.modules.Module;
+import org.jboss.modules.ModuleClassLoader;
+import org.jboss.msc.service.ServiceName;
+import org.wildfly.microprofile.opentracing.smallrye.TracingCDIExtension;
+import org.wildfly.microprofile.opentracing.smallrye.TracingLogger;
+import org.wildfly.microprofile.opentracing.smallrye.WildFlyTracerFactory;
+import org.wildfly.security.manager.WildFlySecurityManager;
+
+public class TracingDeploymentProcessor implements DeploymentUnitProcessor {
+
+    private static final AttachmentKey<Tracer> ATTACHMENT_KEY = AttachmentKey.create(Tracer.class);
+    private static final String SPAN_FINISHING_FILTER = "io.smallrye.opentracing.contrib.jaxrs2.server.SpanFinishingFilter";
```

Comment:
If this constant value is the only diff vs the non-jakarta variant, that constant can be extracted out into a simple Constants class that is forked, instead of forking the complex logic.

Reply:
Done.